### PR TITLE
Fix #1 

### PR DESCRIPTION
**Pull request by LewisTehMinerz**
This reverts the change that switched the library back to Express types (so it now uses the generic `http` types) and fixes the header problem.

Also, this PR gobbles errors that come from `noble-ed25519`. The cause of the error in #5 was due to the bad header read but the actual issue was that the library doesn't error out if it fails to parse, and just silently carries on with a broken state. It'll now just `return false;` if it encounters an error.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -60,28 +60,48 @@ async function verifyKey(
...
       chunks.push(chunk);
     });
     req.on('end', async function () {
+      // header sanity check
+      if (!timestamp || !signature) {
+        res.statusCode = 401;
+        res.end('Invalid headers');
+        return;
+      }
+
```
Comment (typpo) [inline src/index.ts:104]: Let's move this to the top of the function, and we can also get rid of the `|| ''`

**Review by typpo — CHANGES_REQUESTED** [review]
Thanks for figuring this one out!  Just one small request

**Comment by typpo** [comment]
ha, I screwed this up and merged too early as the new logic to handle bodyparser now uses `req.body`

**Comment by LewisTehMinerz** [comment]
oh, I already merged that in I'm pretty sure?

**Pull request merged**